### DISCUSSION TITLE
[Feature Request] Evaluate variables within variables

### DISCUSSION THREAD

**Discussion by surdaft** [grafana/grafana#58412] [Dashboards, Panels & Visualizations]
Related to #58289 

I wonder if we could implement the functionality to use variables within variables. Example:

1. I create a variable for `region` which would be `london`
2. I create a variable for `environment` which would be `dev` or `prod`
3. I create a constant called `prefix` which is `${region}-${environment}` which can be used in queries like so:
    * `prometheus_query{targetgroup=~"${prefix}-api/.+"}` in replacement for `prometheus_query{targetgroup=~"${region}-${environment}-api/.+"}`

---

Additional use cases:

Constant for a "base query" which can have slight alterations within the query:
```
variables:
$region = london
$environment = dev
$prefix = ${region}-${environment}
$query = prometheus_query{targetgroup=~"${prefix}-api/.+"}
$which = avg,min,max,count
$query_over_time = ${which}_over_time($query [15m])

panels:
rate($query [10m])

repeat panels: $query_over_time
```

To reference the `env` and `region` or `prefix` of both within another variable query:
```
label_values(aws_lambda_invocation_sum{=~"${prefix}.+"}, function_name)
```

Also related slack chat:
https://grafana.slack.com/archives/C0Y4TLW74/p1667901860237889

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
